### PR TITLE
Add Forge command for changing entity dimensions

### PR DESCRIPTION
**Pull request by bs2609**
See https://github.com/MinecraftForge/MinecraftForge/pull/4575#issuecomment-352159728.

Pretty basic, especially around placement, so mainly just useful for testing at the moment.

Also tweaks the `PlayerList` patches to remove some hardcoding around the end dimension if a custom teleporter is supplied.

### CODE REVIEW

**Comment by brandon3055** [comment]
You shouldn't be using a Teleporter for this. Those are really only useful for things like custom portals that need to generate a portal at the other end, and handle sending the entity to an existing portal. Hacking them to do something like this is not advisable. 

You do not need to modify any vanilla code to make this work you just need to implement your teleportation code properly. 

I suggest you take a look at my teleportation handler. Its reverse engineered from minecrafts code and it just works. It also handles recursive teleportation of ridden entities and passengers.

https://github.com/brandon3055/BrandonsCore/blob/master/src/main/java/com/brandon3055/brandonscore/lib/TeleportUtils.java

File: src/main/java/net/minecraftforge/server/command/CommandSetDimension.java [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+    @Override
+    public String getName()
+    {
+        return "setdimension";
+    }
+
+    @Override
+    public List<String> getAliases()
+    {
+        return Collections.singletonList("setdim");
```
Comment (mezz) [inline src/main/java/net/minecraftforge/server/command/CommandSetDimension.java:54]: `tp` might be another nice alias for this as well

**Comment by mezz** [comment]
Brandon has a lot of experience in this area, I will mostly defer to them for this review.

Also, it would be nice if you could move the main logic to a helper function so mods can use it to teleport across dimensions properly.

File: src/main/java/net/minecraftforge/server/command/CommandSetDimension.java (hunk 1 above)
Comment (bs2609) [reply src/main/java/net/minecraftforge/server/command/CommandSetDimension.java:54]: I picked a different name to distinguish this from the vanilla `tp` and `teleport` commands, as it's mainly intended for something else.

If it's wanted, I can add more aliases, though.

**Comment by brandon3055** [comment]
Your teleportation code is completely broken. Try teleporting from the end to the overworld. 

Not only does it not clear boss info but the client is completely out of sync with the server. Try spawning an entity and nothing happens until you relog.

Edit: Have not been able to reproduce the desync after my initial test.

**Comment by bs2609** [comment]
This is on hold for #4602 at least. I think it's better for any needed changes to be made elsewhere, and then this PR can be updated to use them.

**Comment by bs2609** [comment]
This has been updated and added to #4602 now.

**Pull request closed without merging**